Tests/DistributedActorsTests/RemoteCallTests.swift

```diff
@@ -0,0 +1,330 @@
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.remoteCall.codableErrorAllowance = .none
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local)
+        let remoteGreeterRef = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let error = try await shouldThrow {
+            _ = try await remoteGreeterRef.helloThrow(codable: true)
+        }
+        guard error is GenericRemoteCallError else {
+            throw TestError("Expected GenericRemoteCallError, got \(error)")
+        }
+    }
+
+    func test_remoteCall_customCodableErrorAllowList_errorInList() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([GreeterCodableError.self, AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([GreeterCodableError.self, AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local)
+        let remoteGreeterRef = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let error = try await shouldThrow {
+            _ = try await remoteGreeterRef.helloThrow(codable: true)
+        }
+        guard error is GreeterCodableError else {
+            throw TestError("Expected GreeterCodableError, got \(error)")
+        }
+    }
+
+    func test_remoteCall_customCodableErrorAllowList_errorNotInList() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
```

Comment:
Hm I'm wondering if this is too annoying... we have to allow list types, that's for sure... and serializers as well... Hm, for now this is okey, maybe some sugar in the future "register trusted error type"?

Reply:
Maybe I still don't get the directions right, but for me it's confusing in that I don't know on which node I need to set the allow list and register types (or maybe there should be `registerInbound` and `registerOutbound`?). I feel like I should only need to do these on the responding node.

> Hm, for now this is okey, maybe some sugar in the future "register trusted error type"?

We can probably make `ClusterSystemSettings` smarter and have it automatically register the allowed types.

Reply:
Hmm, how come `GenericRemoteCallError` doesn't need to be registered?

Reply:
Yeah I agree this is confusing, I'll have a look

Reply:
Follow up ticket https://github.com/apple/swift-distributed-actors/issues/1024